### PR TITLE
kv/rocksdb: dump transactions on error

### PR DESCRIPTION
**Pull request by somnathr**
Some verbose BlueStore logging very much needed for debugging crash.

### CODE REVIEW

**Comment by somnathr** [comment]
@liewegas I know you are busy , but, if you can take a look at this and merge my debugging will be much easier. Thanks !

File: src/kv/RocksDBStore.h [hunk 1]
```diff
@@ -137,6 +137,98 @@ class RocksDBStore : public KeyValueDB {
...
+  virtual void SingleDelete(const rocksdb::Slice& key) override {
+
+    string prefix ((key.ToString()).substr(0,1));
+    string key_to_decode ((key.ToString()).substr(2,string::npos));
+
+    seen += "\nSingleDelete(Prefix = "+ prefix + " Key = " 
+          + pretty_binary_string(key_to_decode) + ")";
+    num_seen++;
+  }
+  virtual void Delete(const rocksdb::Slice& key) override {
```
Comment (liewegas) [inline src/kv/RocksDBStore.h]: can you fix the indentation?  the section above this looks off.  thanks!

**Comment by liewegas** [comment]
otherwise lgtm

File: src/kv/RocksDBStore.h (hunk 1 above)
Comment (somnathr) [inline src/kv/RocksDBStore.h]: @liewegas meaning removing the blank spaces on ln 201, 204 ?

**Comment by liewegas** [comment]
maybe it's github, but it looks like the preceding hundred or so lines is 
2 spaces too far to the left?

**Comment by liewegas** [comment]
https://github.com/somnathr/ceph/blob/9338c5d0ce9dee5e815dceeec100d293212f6421/src/kv/RocksDBStore.h#L156-L209

**Comment by somnathr** [comment]
Yes, you are right , fixing that..

**Comment by somnathr** [comment]
@liewegas done formatting , please have a look.

**Pull request merged**